apis/externalsecrets/v1alpha1/secretstore_vault_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+type VaultKVStoreVersion string
+
+const (
+	VaultKVStoreV1 VaultKVStoreVersion = "v1"
+	VaultKVStoreV2 VaultKVStoreVersion = "v2"
+)
+
+// Configures an store to sync secrets using a HashiCorp Vault
+// KV backend.
+type VaultProvider struct {
+	// Auth configures how secret-manager authenticates with the Vault server.
+	Auth VaultAuth `json:"auth"`
+
+	// Server is the connection address for the Vault server, e.g: "https://vault.example.com:8200".
+	Server string `json:"server"`
+
+	// Path is the mount path of the Vault KV backend endpoint, e.g:
+	// "secret". The v2 KV secret engine version specific "/data" path suffix
+	// for fetching secrets from Vault is optional and will be appended
+	// if not present in specified path.
+	Path string `json:"path"`
+
+	// Version is the Vault KV secret engine version. This can be either "v1" or
+	// "v2". Version defaults to "v2".
```

Comment:
hey @mcavoyk, while testing locally, I found that the default value here doesn't work, returning the following validation error:
```
error: error validating "vault-secret-store.yaml": error validating data: ValidationError(SecretStore.spec.provider.vault): missing required field "version" in io.external-secrets.v1alpha1.SecretStore.spec.provider.vault; if you choose to ignore these errors, turn valid
```

I found out you also need to add:

```
// +kubebuilder:validation:Optional
```

What do u think?

Reply:
What version of Kubernetes, it may require v1.17 for the default to work properly. On kube < 1.17 the value would need to be provided explicitly. I'll take a look and maybe we can make this optional for this case.

Reply:
I'm running v1.17.0 on Kind.